### PR TITLE
Relax sampleqc on symbolic alleles

### LINKED ISSUE #3413: "Invalid Allele" message when alt ref is  <DEL>
To report a bug, fill in the information below. 
For support and feature requests, please use the discussion forum: http://discuss.hail.is/

-------------------------------------------------------------------------------------------

### What you did:
Tried to run sample_qc on mt table imported from Delly vcf. 
hl.sample_qc(ds)

### What went wrong (all error messages here, including the full java stack trace):

The sample_qc had a problem when alt ref is <DEL>

```
[Stage 3:>                                                      (0 + 140) / 415]Traceback (most recent call last):
  File "/restricted/projectnb/genpro/github/hail.qc/delly-qc.py", line 35, in <module>
    ds = hl.sample_qc(ds)
  File "<decorator-gen-902>", line 2, in sample_qc
  File "/restricted/projectnb/genpro/github/hail/python/hail/typecheck/check.py", line 490, in _typecheck
    return __orig_func__(*args_, **kwargs_)
  File "/restricted/projectnb/genpro/github/hail/python/hail/methods/qc.py", line 91, in sample_qc
    return MatrixTable(Env.hail().methods.SampleQC.apply(require_biallelic(dataset, 'sample_qc')._jvds, name))
  File "/share/pkg/spark/2.2.0/install/python/lib/py4j-0.10.4-src.zip/py4j/java_gateway.py", line 1133, in __call__
  File "/restricted/projectnb/genpro/github/hail/python/hail/utils/java.py", line 196, in deco
    'Error summary: %s' % (deepest, full, hail.__version__, deepest)) from None
hail.utils.java.FatalError: HailException: invalid allele "<DEL>"

Java stack trace:
org.apache.spark.SparkException: Job aborted due to stage failure: Task 2 in stage 3.0 failed 4 times, most recent failure: Lost task 2.3 in stage 3.0 (TID 160, scc-q01.scc.bu.edu, executor 4): is.hail.utils.HailException: invalid allele "<DEL>"
        at is.hail.utils.ErrorHandling$class.fatal(ErrorHandling.scala:6)
        at is.hail.utils.package$.fatal(package.scala:26)
        at is.hail.variant.AltAlleleMethods$.validate(AltAlleleMethods.scala:24)
        at is.hail.variant.AltAlleleMethods$.altAlleleType(AltAlleleMethods.scala:29)
        at is.hail.methods.SampleQCCombiner$.alleleIndices(SampleQC.scala:44)
        at is.hail.methods.SampleQC$$anonfun$results$1$$anonfun$apply$1.apply(SampleQC.scala:178)
        at is.hail.methods.SampleQC$$anonfun$results$1$$anonfun$apply$1.apply(SampleQC.scala:175)
        at scala.collection.Iterator$class.foreach(Iterator.scala:893)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
        at is.hail.methods.SampleQC$$anonfun$results$1.apply(SampleQC.scala:175)
        at is.hail.methods.SampleQC$$anonfun$results$1.apply(SampleQC.scala:170)
        at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:797)
        at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:797)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
        at org.apache.spark.scheduler.Task.run(Task.scala:108)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:335)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

Driver stacktrace:
        at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1499)
        at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1487)
        at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1486)
        at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
        at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1486)
        at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:814)
        at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:814)
        at scala.Option.foreach(Option.scala:257)
        at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:814)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1714)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1669)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1658)
        at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
        at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:630)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2022)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2119)
        at org.apache.spark.rdd.RDD$$anonfun$reduce$1.apply(RDD.scala:1026)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
        at org.apache.spark.rdd.RDD.withScope(RDD.scala:362)
        at org.apache.spark.rdd.RDD.reduce(RDD.scala:1008)
        at org.apache.spark.rdd.RDD$$anonfun$treeAggregate$1.apply(RDD.scala:1151)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
        at org.apache.spark.rdd.RDD.withScope(RDD.scala:362)
        at org.apache.spark.rdd.RDD.treeAggregate(RDD.scala:1128)
        at org.apache.spark.rdd.RDD$$anonfun$treeReduce$1.apply(RDD.scala:1059)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
        at org.apache.spark.rdd.RDD.withScope(RDD.scala:362)
        at org.apache.spark.rdd.RDD.treeReduce(RDD.scala:1037)
        at is.hail.methods.SampleQC$.results(SampleQC.scala:206)
        at is.hail.methods.SampleQC$.apply(SampleQC.scala:221)
        at is.hail.methods.SampleQC.apply(SampleQC.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:280)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:214)
        at java.lang.Thread.run(Thread.java:745)is.hail.utils.HailException: invalid allele "<DEL>"
        at is.hail.utils.ErrorHandling$class.fatal(ErrorHandling.scala:6)
        at is.hail.utils.package$.fatal(package.scala:26)
        at is.hail.variant.AltAlleleMethods$.validate(AltAlleleMethods.scala:24)
        at is.hail.variant.AltAlleleMethods$.altAlleleType(AltAlleleMethods.scala:29)
        at is.hail.methods.SampleQCCombiner$.alleleIndices(SampleQC.scala:44)
        at is.hail.methods.SampleQC$$anonfun$results$1$$anonfun$apply$1.apply(SampleQC.scala:178)
        at is.hail.methods.SampleQC$$anonfun$results$1$$anonfun$apply$1.apply(SampleQC.scala:175)
        at scala.collection.Iterator$class.foreach(Iterator.scala:893)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
        at is.hail.methods.SampleQC$$anonfun$results$1.apply(SampleQC.scala:175)
        at is.hail.methods.SampleQC$$anonfun$results$1.apply(SampleQC.scala:170)
        at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:797)
        at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:797)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
        at org.apache.spark.scheduler.Task.run(Task.scala:108)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:335)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)


Hail version: devel-63d60cc
Error summary: HailException: invalid allele "<DEL>"

```

### PR DESCRIPTION
**Pull request by tpoterba**
fixes #3413

### CODE REVIEW

File: src/main/scala/is/hail/variant/AltAlleleMethods.scala [hunk 1]
```diff
@@ -25,9 +25,9 @@ object AltAlleleMethods {
   }
 
   def altAlleleType(ref: String, alt: String): AltAlleleType = {
-    validate(ref)
-    validate(alt)
-    if (ref.length == alt.length) {
+    if (!alleleRegex.matches(ref) || !alleleRegex.matches(alt))
```
Comment (jbloom22) [inline src/main/scala/is/hail/variant/AltAlleleMethods.scala]: I’d pull out the ! and change || to &&

**Comment by tpoterba** [comment]
Given the drama about the followup PR, I'm reopening this to unblock @jjfarrell

**Comment by tpoterba** [comment]
is this worth the minute of time + context switch? if so, I'll do it

**Comment by tpoterba** [comment]
I'd argue this is a nicer UX - Having an "invalid" or "unknown" type lets people with weird alleles (and people do have weird alleles) actually run their pipelines instead of erroring out.

**Comment by tpoterba** [comment]
mostly I just want to unblock people until we move this / splitmulti / sampleqc entirely over to python

**Comment by tpoterba** [comment]
`hl.allele_type` could also return missing for invalid input. But I don't want to refactor all the scala if we're just going to remove it

**Pull request closed without merging**